execution/executor/src/components/in_memory_state_calculator.rs

```diff
@@ -25,25 +25,43 @@ use std::{
     sync::Arc,
 };
 use storage_interface::{
-    in_memory_state::InMemoryState, verified_state_view::StateCache, DbReader, TreeState,
+    in_memory_state::InMemoryState,
+    verified_state_view::{StateCache, VerifiedStateView},
+    DbReader, TreeState,
 };
 
 pub trait IntoLedgerView {
     fn into_ledger_view(self, db: &Arc<dyn DbReader>) -> Result<ExecutedTrees>;
 }
 
 impl IntoLedgerView for TreeState {
-    fn into_ledger_view(self, _db: &Arc<dyn DbReader>) -> Result<ExecutedTrees> {
-        let checkpoint_num_txns = self.num_transactions;
+    fn into_ledger_view(self, db: &Arc<dyn DbReader>) -> Result<ExecutedTrees> {
+        let checkpoint_num_txns = self
+            .num_transactions
+            .checked_sub(self.write_sets_after_checkpoint.len() as LeafCount)
+            .ok_or_else(|| anyhow!("State checkpoint pre-dates genesis."))?;
         let checkpoint =
-            InMemoryState::new_at_checkpoint(self.state_root_hash, checkpoint_num_txns);
-
+            InMemoryState::new_at_checkpoint(self.state_checkpoint_hash, checkpoint_num_txns);
+
+        let checkpoint_state_view = VerifiedStateView::new(
+            StateViewId::Miscellaneous,
+            db.clone(),
+            checkpoint_num_txns.checked_sub(1),
```

Comment:
This is a bit annoying and confusing IMO, we generally deal with txn version throughout our code base - adding number of txns, which is 1 more than the version is not consistent with rest of the code base. Any way we can keep version here instead?